cla-backend-go/signatures/repository.go

```diff
@@ -939,6 +943,115 @@ func (repo repository) GetProjectCompanySignatures(params signatures.GetProjectC
+
+	builder := expression.NewBuilder().WithProjection(buildProjection())
+	var filter expression.ConditionBuilder
+	var filterAdded bool
+
+	// Filter condition to cater for approved and signed signatures
+	signatureApprovedExpression := expression.Name("signature_approved").Equal(expression.Value(true))
+	filter = addConditionToFilter(filter, signatureApprovedExpression, &filterAdded)
+
+	signatureSignedExpression := expression.Name("signature_signed").Equal(expression.Value(true))
+	filter = addConditionToFilter(filter, signatureSignedExpression, &filterAdded)
+
+	if filterAdded {
+		builder = builder.WithFilter(filter)
+	}
+	builder = builder.WithKeyCondition(condition)
+
+	// Use the nice builder to create the expression
+	expr, err := builder.Build()
+	if err != nil {
+		log.Warnf("error building expression for project signature query, projectID: %s, error: %v",
+			projectID, err)
+		return nil, err
+	}
+
+	// Assemble the query input parameters
+	queryInput := &dynamodb.QueryInput{
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		KeyConditionExpression:    expr.KeyCondition(),
+		ProjectionExpression:      expr.Projection(),
+		FilterExpression:          expr.Filter(),
+		TableName:                 aws.String(tableName),
+		IndexName:                 aws.String(indexName), // Name of a secondary index to scan
+	}
+
+	results, errQuery := repo.dynamoDBClient.Query(queryInput)
+
+	if errQuery != nil {
+		log.Warnf("error retrieving project signature ID for project: %s, error: %v",
+			projectID, errQuery)
+		return nil, errQuery
+	}
+
+	log.Debugf("Signature project query took: %v resulting in %d results",
+		utils.FmtDuration(time.Since(queryStartTime)), len(results.Items))
+
+	// Convert the list of DB models to a list of response models
+	signatures, modelErr := repo.buildProjectSignatureModels(results, projectID)
+	if modelErr != nil {
+		log.Warnf("error converting DB model to response model for signatures with project %s, error: %v",
+			projectID, modelErr)
+		return nil, modelErr
+	}
+
+	return &models.Signatures{
+		ProjectID:  projectID,
+		Signatures: signatures,
+	}, nil
+}
+
+func (repo repository) RemoveProject(signatureID string) error {
+	// Update project signatures for signature_approved and notes attributes
+	signatureTableName := fmt.Sprintf("cla-%s-signatures", repo.stage)
+
+	expressionAttributeNames := map[string]*string{}
+	expressionAttributeValues := map[string]*dynamodb.AttributeValue{}
+	updateExpression := "SET "
+
+	expressionAttributeNames["#A"] = aws.String("signature_approved")
+	expressionAttributeValues[":a"] = &dynamodb.AttributeValue{BOOL: aws.Bool(false)}
+	updateExpression = updateExpression + " #A = :a"
+
+	expressionAttributeNames["#S"] = aws.String("note")
+	expressionAttributeValues[":s"] = &dynamodb.AttributeValue{S: aws.String("Signature approved set to false because of project: %s deletion")}
```

Comment:
Two things:
- Maybe say: "Signature invalidated (approved set to false) due to CLA Group/Project: %s deletion"
- You need to include the project name via: `aws.String(fmt.Sprintf("Signature invalidated (approved set to false) due to CLA Group/Project: %s deletion", projectName))`